README.md

```diff
@@ -32,415 +32,450 @@
+### <a name='definedpseudo-class'></a>`:defined` pseudo-class
+The `:defined` pseudo-class applies to all elements that are known to the user agent, i.e. it applies to all elements that are either built-in or successfully upgraded to custom elements.
 
-This for example allows for hiding the non-functional custom elements in CSS if the element definitions for the non-essential elements are being downloaded asynchronously to improve the initial load.
+This allows, for example, for hiding any non-functional custom elements, while the element definitions for the non-essential elements are being downloaded asynchronously to reduce the initial load time.
 
 ```css
-my-element:not(:defined) {
+my-custom-element:not(:defined) {
 	display: block;
 	height: 100vh;
 	opacity: 0;
 }
 ```
 
-The :defined pseudo class unfortunately can not be polyfilled, so it will only work for user agents that implement the custom element v1 specifications natively. However, you can mimic this behaviour with for example an attribute that is removed once the element is connected.
+Note that the :defined pseudo class cannot be polyfilled, so it will only work for user agents that implement the custom element v1 specifications natively.
+However, you can mimic this behaviour with, for example, an attribute that is removed once the element is connected.
 
 ```css
-my-element[unresolved] {
+my-custom-element[unresolved] {
 	display: block;
 	height: 100vh;
 	opacity: 0;
 }
 ```
 
 ```javascript
-class MyElement extends HTMLElement {
-	...
+class MyCustomElement extends HTMLElement {
+	// ...
 	connectedCallback() { 
 		this.removeAttribute('unresolved');
 	}
-	...
+	// ...
 }
 ```
 
 ```html
-<my-element unresolved></my-element>
+<my-custom-element unresolved></my-custom-element>
 ```
 
 ## <a name='Custombuilt-inelementscurrentlyunsupported'></a>Custom built-in elements (currently unsupported)
-* An element extending a built in element is also known as a *customized built-in element*.
-* It is intended to work exactly like an autonomous custom element, however inheriting all behavior that the element that is extended would have.
-* In stead of extending `HTMLElement`, to create a customized built-in element it must extend a specific subclass like `HTMLButtonElement`, and supply an object specifying what element it extends as multiple elements use the same interface classes.
+* An element extending a built-in element is known as a *customized built-in element*.
+* It is intended to work exactly like an autonomous custom element, but inheriting all behavior from its parent built-in element.
+* Rather than extending `HTMLElement`, a customized built-in element it should extend one of its subclasses (like `HTMLButtonElement`), and supply a configuration object to specify what element is extended (since multiple elements can use the same interface class).
```

Comment:
Maybe rephrase "it should extend" to "extends" and fix the rest of the sentence accordingly